R/register_handlers.R

```diff
@@ -0,0 +1,111 @@
+#' Register handlers for logging messages, warnings and errors
+#'
+#' @param namespace (`character(1)`) the logger namespace
+#' @param package (`character(1)`) the package name
+#'
+#' @return `NULL` invisibly. Called for its side effects.
+#'
+#' @details This function registers global handlers for messages, warnings and errors.
+#' The handlers will investigate the call stack and if it contains a function
+#' from the package, the message, warning or error will be logged into the respective
+#' namespace.
+#' The handlers are registered only once per package and type. Consecutive calls will be ignored.
+#'
+#' @note Registering handlers is disallowed within `tryCatch` blocks which is used throughout
+#' base R and other packages. Because of this, the `register_handlers` function is wrapped in a
+#' `try` block to avoid errors.
+#'
+#' @seealso [globalCallingHandlers()]
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' register_logger(namespace = "teal.logger")
+#' register_handlers("teal.logger")
+#' # see the outcome
+#' globalCallingHandlers()
+#' }
+register_handlers <- function(namespace, package = namespace) {
+  try(
+    {
+      register_handler_message(namespace = namespace, package = package)
+      register_handler_warning(namespace = namespace, package = package)
+      register_handler_error(namespace = namespace, package = package)
+    },
+    silent = TRUE
+  )
+
+  invisible(NULL)
+}
+
+#' @keywords internal
+register_handler_message <- function(namespace, package) {
+  register_handler_type(namespace, package, "message")
+}
+#' @keywords internal
+register_handler_warning <- function(namespace, package) {
+  register_handler_type(namespace, package, "warning")
+}
+#' @keywords internal
+register_handler_error <- function(namespace, package) {
+  register_handler_type(namespace, package, "error")
+}
```

Comment:
@pawelru why do we need those 3 functions? can't we use `register_handler_type` 3 times in `register_handler` instead of having extra 3 functions called once?

Reply:
ok will remove them